#! /bin/sh /usr/share/dpatch/dpatch-run
## firefox_iceweasel_rename.dpatch by  <ct@piglets.com>
##
## DP: The OpenSearch plugin searches for firefox search plugins,
## DP: but of course, on a Debian system this is to be found under
## DP: iceweasel, not firefox.
## DP:
## DP: This patch also makes the Firefox bookmarks plugin index
## DP: Iceweasel bookmarks.

@DPATCH@
Index: gnome-do-plugins/Firefox/src/PlacesItemSource.cs
===================================================================
--- gnome-do-plugins.orig/Firefox/src/PlacesItemSource.cs	2011-02-23 22:28:24.781232052 +1100
+++ gnome-do-plugins/Firefox/src/PlacesItemSource.cs	2011-02-24 10:47:31.564914977 +1100
@@ -161,8 +161,13 @@
 			profile = null;
 			home = Environment.GetFolderPath (Environment.SpecialFolder.Personal);
 			
-			path = Path.Combine (home, ".mozilla/firefox/profiles.ini");
-			using (StreamReader r = File.OpenText (path)) {
+			path = Path.Combine (home, ".mozilla");
+			if (Directory.Exists (Path.Combine (path, "firefox"))) {
+				path = Path.Combine (path, "firefox");
+			} else if (Directory.Exists (Path.Combine (path, "iceweasel"))) {
+				path = Path.Combine (path, "iceweasel");
+			}
+			using (StreamReader r = File.OpenText (Path.Combine (path, "profiles.ini"))) {
 				while ((line = r.ReadLine ()) != null) {
 					if (line.StartsWith (BeginDefaultProfile)) {
 						break;
@@ -173,7 +178,7 @@
 					}
 				}
 			}
-			return new [] {home, ".mozilla", "firefox", profile}.Aggregate (Path.Combine); 
+			return Path.Combine (path, profile); 
 		}
 
 
Index: gnome-do-plugins/OpenSearch/src/FirefoxOpenSearchDirectoryProvider.cs
===================================================================
--- gnome-do-plugins.orig/OpenSearch/src/FirefoxOpenSearchDirectoryProvider.cs	2011-02-23 22:28:24.801232770 +1100
+++ gnome-do-plugins/OpenSearch/src/FirefoxOpenSearchDirectoryProvider.cs	2011-02-24 10:47:31.594916063 +1100
@@ -37,6 +37,9 @@
 	/// </summary>
 	public class FirefoxOpenSearchDirectoryProvider
 	{
+		private static string firefoxPath = "/usr/bin/firefox";
+		private static string iceweaselPath = "/usr/bin/iceweasel";
+		
 		private List<string> openSearchPluginDirectories;
 		
 		/// <summary>
@@ -88,11 +91,17 @@
 				// for installing in different directories. We could certainly shell
 				// out and call which or something...
 				string beginLibDir = "LIBDIR=";
-				string binFile = "/usr/bin/firefox";
+				string binFile = "";
 				string line, libDir;		
 				
 				libDir = null;
 				
+				if (File.Exists (firefoxPath)) {
+					binFile = firefoxPath;
+				} else if (File.Exists (iceweaselPath)) {
+					binFile = iceweaselPath;
+				}
+				
 				using (StreamReader r = File.OpenText (binFile)) {
 					while (null != (line = r.ReadLine ())) {
 						if (line.StartsWith (beginLibDir)) {
@@ -117,6 +126,18 @@
 			return null;
 		}
 		
+		private string UserBrowserPath {
+			get {
+				string mozillaPath = Path.Combine (Environment.GetFolderPath (Environment.SpecialFolder.Personal), ".mozilla");
+				if (Directory.Exists (Path.Combine (mozillaPath, "firefox"))) {
+				    mozillaPath = Path.Combine (mozillaPath, "firefox");
+				} else if (Directory.Exists (Path.Combine (mozillaPath, "iceweasel"))) {
+					mozillaPath = Path.Combine (mozillaPath, "iceweasel");
+				}
+				return mozillaPath;
+			}
+		}
+			
 		/// <summary>
 		/// Retrieves the profile plugin directory, which is where the 
 		/// user installed OpenSearch plugins are located.
@@ -134,7 +155,7 @@
 				
 				profile = null;				
 				
-				profilePath = Path.Combine (Environment.GetFolderPath (Environment.SpecialFolder.Personal), ".mozilla/firefox/profiles.ini");
+				profilePath = Path.Combine (UserBrowserPath, "profiles.ini");
 				using (StreamReader r = File.OpenText (profilePath)) {
 					while (null != (line = r.ReadLine ())) {
 						if (line.StartsWith (beginDefaultProfile)) break;
@@ -147,7 +168,7 @@
 				}
 							
 				if(profile != null) {
-					string path = Path.Combine (Environment.GetFolderPath (Environment.SpecialFolder.Personal), ".mozilla/firefox");
+					string path = UserBrowserPath;
 					path = Path.Combine (path, profile);
 					path = Path.Combine (path, "searchplugins");
 				
Index: gnome-do-plugins/po/POTFILES.skip
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ gnome-do-plugins/po/POTFILES.skip	2011-02-24 10:48:49.917749384 +1100
@@ -0,0 +1,3 @@
+# dh_auto_test runs intltool-extract, which bombs because this isn't
+# in POTFILES.in.  Bah!
+.pc/01_firefox_iceweasel_rename.patch/Firefox/src/PlacesItemSource.cs
